app/components/autocomplete/date_suggestion/index.js

```diff
@@ -14,10 +15,13 @@ function mapStateToProps(state, ownProps) {
 
     const newValue = value.substring(0, cursorPosition);
     const matchTerm = getMatchTermForDateMention(newValue);
+    const serverVersion = state.entities.general.serverVersion;
+    const enableDateSuggestion = isMinimumServerVersion(serverVersion, 5, 2);
```

Comment:
I don't think you need this here cause you are not rendering this component if the `enableDateSuggestion` prop is false in the autocomplete component